testsuite/tests/typing-jkind-bounds/subsumption/quality.ml

```diff
@@ -0,0 +1,286 @@
+  type t = a
+  type u = a
+end
+
+module M : sig
+  type t : value mod portable uncontended with M.u
+end = struct
+  type t : value mod portable
+end
+[%%expect {|
+module type S = sig type t type u = t end
+type a
+module M : sig type u = a end
+Lines 14-16, characters 6-3:
+14 | ......struct
+15 |   type t : value mod portable
+16 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t : value mod portable end
+       is not included in
+         sig type t : value mod uncontended portable with M.u end
+       Type declarations do not match:
+         type t : value mod portable
+       is not included in
+         type t : value mod uncontended portable with M.u
+       The kind of the first is value mod portable
+         because of the definition of t at line 15, characters 2-29.
+       But the kind of the first must be a subkind of
+         value mod uncontended portable with M.u
+         because of the definition of t at line 13, characters 2-50.
+|}]
+
+module M : sig
+  type a = [`a of string | `b]
+  type t : value mod global with a
+end = struct
+  type a = [`a of string | `b]
+  type t
+end
+[%%expect {|
+Lines 4-7, characters 6-3:
+4 | ......struct
+5 |   type a = [`a of string | `b]
+6 |   type t
+7 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type a = [ `a of string | `b ] type t end
+       is not included in
+         sig
+           type a = [ `a of string | `b ]
+           type t : value mod global with a
+         end
+       Type declarations do not match:
+         type t
+       is not included in
+         type t : value mod global with a
+       The kind of the first is value
+         because of the definition of t at line 6, characters 2-8.
+       But the kind of the first must be a subkind of value mod global with a
+         because of the definition of t at line 3, characters 2-34.
+|}]
+
+module M : sig
+  type a = < value : string >
+  type t : value mod global with a
+end = struct
+  type a = < value : string >
+  type t
+end
+[%%expect {|
+Lines 4-7, characters 6-3:
+4 | ......struct
+5 |   type a = < value : string >
+6 |   type t
+7 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type a = < value : string > type t end
+       is not included in
+         sig type a = < value : string > type t : value mod global with a end
+       Type declarations do not match:
+         type t
+       is not included in
+         type t : value mod global with a
+       The kind of the first is value
+         because of the definition of t at line 6, characters 2-8.
+       But the kind of the first must be a subkind of value mod global with a
+         because of the definition of t at line 3, characters 2-34.
+|}]
+
+(* CR layouts v2.8: gadts shouldn't be "best" because we intend to give them more refined
+   jkinds in the future. So this program will error in the future. *)
+type gadt = Foo : int -> gadt
```

Comment:
Would be cool to repeat this test with `[@@unboxed]`.

Reply:
done